src/wallets/WalletContractV5.ts

```diff
@@ -0,0 +1,310 @@
+    }
+
+    private createActions( args: {  messages: MessageRelaxed[], sendMode?: Maybe<SendMode> }) {
+        const sendMode = args.sendMode ?? SendMode.PAY_GAS_SEPARATELY + SendMode.IGNORE_ERRORS;
+        const actions: OutActionSendMsg[] = args.messages.map(message => ({ type: 'sendMsg', mode: sendMode, outMsg: message}));
+        return actions;
+    }
+
+    /**
+     * Create signed transfer
+     */
+    createTransfer(args: Wallet5SendArgs & { messages: MessageRelaxed[] }) {
+        const { messages, ...rest } = args;
+        return this.createRequest({
+            ...rest,
+            actions: this.createActions({ messages, sendMode: args.sendMode })
+        })
+    }
+
+    /**
+     * Create signed transfer async
+     */
+    createTransferAndSignRequestAsync(args: ExternallySingedAuthWallet5SendArgs & { messages: MessageRelaxed[] }) {
+        const { messages, ...rest } = args;
+        return this.createAndSignRequestAsync({
+            ...rest,
+            actions: this.createActions({ messages, sendMode: args.sendMode })
+        })
+    }
+
+    /**
+     * Create signed add extension request
+     */
+    createAddExtension(args: Wallet5SendArgs & { extensionAddress: Address }) {
+        const { extensionAddress, ...rest } = args;
+        return this.createRequest({
+            actions: [{
+                type: 'addExtension',
+                address: extensionAddress
+            }],
+            ...rest
+        })
+    }
+
+    /**
+     * Create signed remove extension request
+     */
+    createRemoveExtension(args: Wallet5SendArgs & { extensionAddress: Address }) {
+        const { extensionAddress, ...rest } = args;
+        return this.createRequest({
+            actions: [{
+                type: 'removeExtension',
+                address: extensionAddress
+            }],
+            ...rest
+        })
+    }
+
+    /**
+     * Create signed request or extension auth request
+     */
+    createRequest(args: Wallet5SendArgs & { actions: (OutActionSendMsg | OutActionExtended)[] }) {
+        if (args.authType === 'extension') {
+            return createWalletTransferV5ExtensionAuth({
+                ...args,
+                sendMode: args.sendMode ?? SendMode.PAY_GAS_SEPARATELY + SendMode.IGNORE_ERRORS,
+                walletId: storeWalletId(this.walletId)
+            })
+        }
+
+        return createWalletTransferV5SignedAuth({
+            ...args,
+            sendMode: args.sendMode ?? SendMode.PAY_GAS_SEPARATELY + SendMode.IGNORE_ERRORS,
+            walletId: storeWalletId(this.walletId)
+        })
+    }
+
+    /**
+     * Create asynchronously signed request
+     */
+    createAndSignRequestAsync(args: ExternallySingedAuthWallet5SendArgs & { actions: (OutActionSendMsg | OutActionExtended)[] }) {
+        return createWalletTransferV5SignedAuth({
+            ...args,
+            sendMode: args.sendMode ?? SendMode.PAY_GAS_SEPARATELY + SendMode.IGNORE_ERRORS,
```

Comment:
Is sendMode actually used here? I thought this is a part of `actions`

Reply:
fixed